62_CAD/MSDFs.cpp

```diff
@@ -0,0 +1,464 @@
+#include "DrawResourcesFiller.h"
+#include "MSDFs.h"
+#include "nbl/ext/TextRendering/TextRendering.h"
+
+// extents is the size of the MSDF that is generated (probably 32x32)
+// glyphExtents is the area of the "image" that msdf will consider (used as resizing, for fill patterns should be 8x8)
+DrawResourcesFiller::MsdfTextureUploadInfo generateMsdfForShape(std::vector<CPolyline>&& polylines, msdfgen::Shape glyph, uint32_t2 msdfExtents, uint32_t2 glyphExtents)
+{
+	uint32_t glyphW = msdfExtents.x;
+	uint32_t glyphH = msdfExtents.y;
+
+	if (polylines.empty())
+	{
+		return {
+			.cpuBuffer = nullptr,
+			.bufferOffset = 0ull,
+			.imageExtent = { glyphW, glyphH, 1 },
+			.polylines = std::move(polylines),
+		};
+	}
+
+	auto shapeBounds = glyph.getBounds();
+
+	msdfgen::edgeColoringSimple(glyph, 3.0); // TODO figure out what this is
+	msdfgen::Bitmap<float, 4> msdfMap(glyphW, glyphH);
+
+	float scaleX = (1.0 / float(glyphExtents.x)) * glyphW;
+	float scaleY = (1.0 / float(glyphExtents.y)) * glyphH;
+	msdfgen::generateMTSDF(msdfMap, glyph, MsdfPixelRange, { scaleX, scaleY }, msdfgen::Vector2(0.0, 0.0));
+
+	// TODO: Optimize this
```

Comment:
as per discussions in discord, you can remove the for loops and set srcFormat in image upload utility to RGBA8_SFLOAT to get the conversion while memcpying to staging memory with our image filters (all happening inside the util)

Reply:
Done